cfgmgr/nbrmgr.cpp

```diff
@@ -64,6 +65,20 @@ NbrMgr::NbrMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, con
                               TableConsumable::DEFAULT_POP_BATCH_SIZE, default_orch_pri);
     auto consumer = new Consumer(consumerStateTable, this, APP_NEIGH_RESOLVE_TABLE_NAME);
     Orch::addExecutor(consumer);
+          
+    string swtype;
+    Table cfgDeviceMetaDataTable(cfgDb, CFG_DEVICE_METADATA_TABLE_NAME);
+    if(cfgDeviceMetaDataTable.hget("localhost", "switch_type", swtype))
+    {
+        //If this is voq system, let the neighbor manager subscribe to state of SYSTEM_NEIGH
+        //entries. This is used to program static neigh and static route in kernel for remote neighbors.
+        if(swtype == "voq")
+        {
+            string tableName = STATE_SYSTEM_NEIGH_TABLE_NAME;
```

Comment:
where we are populating STATE_SYSTEM_NEIGH_TABLE_NAME ? Is it in another PR ?

Reply:
We are populating this table in neighorch.cpp (doVoqSystemNeighTask()) in this PR #1431 only. For the remote neighbors, we need to add static neighbor and static route in the kernel. Since it was recommended not to add the kernel entries from orchagent, we program kernel entries from the nbrmgr (as per @prsunny review comment). In neighorch, after programming the remote neighbor in SAI, we add an entry in state db in STATE_SYSTEM_NEIGH_TABLE to trigger programming kernel entries here.